lib/matplotlib/tests/test_axes.py

```diff
@@ -4972,3 +4972,32 @@ def test_bar_single_height():
     ax.bar(range(4), 1)
     # Check that a horizontal chart with one width works
     ax.bar(0, 1, bottom=range(4), width=1, orientation='horizontal')
+	
+def test_tick_label_selection():
+    #test feature #8128   
+    pd = pytest.importorskip('pandas')
+    # all posibilities for the parameter we are testing
+    which = {None,'minor','major','both'}
+	
+    for w in which:
+        fig ,ax = plt.subplots()
+        df = pd.DataFrame(data=[1] * 5,
+                     index=pd.date_range(start='2017-01-01', periods=5),
+                     columns=['A'])
+        df.plot(ax=ax)
+	
+        #format the tick labels with the given parameter value
+        fig.autofmt_xdate(0.2,30,'right',w)
+	
+	    #stores a boolean of whether or not the minor and major axis should be rotated
+        is_rotated_minor = (w == 'both') or (w == 'minor')
+        is_rotated_major = (w == 'both') or (w == 'major') or (w == None)
+
+	    #check if the major axis were formatted (or not) as expected
+        for label in fig.axes[0].get_xticklabels(False,'major'):
+            print(label.get_rotation())
+            assert (label.get_rotation() == 30) == is_rotated_major
```

Comment:
Oops! That print was only meant to be there for debugging when I was running the test on my own. Would removing it be good enough? The test for the minor labels below has the same format and I guess it's acceptable, right?